text/0000-rust-spec.md

```diff
@@ -0,0 +1,171 @@
+
+# Goals and non-goals
+
+- The goal of the Rust specification work is the creation of a document, the Rust specification.
+
+- The goal is _not_ to change how the language evolves;
+  the relevant teams (Language, Libs-API, …) remain in charge of the evolution of their respective parts of Rust,
+  and will continue to use processes as they see fit (e.g. RFCs).
+
+- The specification will only be considered "official" once the relevant teams have approved of its contents.
+  Changes to the official specification must be approved by the relevant team(s).
+
+- The goal is to serve the needs of Rust users, such as
+  authors of unsafe Rust code, those working on safety critical Rust software,
+  language designers, maintainers of Rust tooling, and so on.
+
+- It is _not_ a primary goal of the specification to aid in the development of alternative Rust implementations,
+  although authors of alternative compilers might still find the specification to be useful.
+
+  What this means is that, unlike the C or C++ standard,
+  the Rust specification does not provide a set of requirements for a compiler to be able to call itself "a Rust™ compiler".
+  Instead, it specifies the behaviour of the Rust compiler.
+  (So, not "A Rust implementation should …", but instead "Rust will …".)
+
+- The scope remains to be determined, but at least includes all topics currently included in 
+  the (incomplete) [Rust Reference](https://doc.rust-lang.org/stable/reference/).
+
+- The Rust specification is expected to replace the current Rust Reference.
+
+- The scope of the specification can grow over time, depending on the needs of Rust users and
+  time and motivation of those working on it.
+
+  For example, it might grow over time to also specify details of Cargo,
+  compiler flags, procedural macros, or other parts that we might initially consider out of scope.
+
+- The specification is specific to the latest version of Rust.
+  A copy of the specification is included with each Rust release,
+  as we currently already do with much of our documentation.
+
+  While the specification might include notes about the Rust version that a feature was introduced in
+  for informative purposes (similar to standard library documentation),
+  it does not attempt to accurately specify older, unsupported versions of Rust.
+
+- The specification specifies all Rust _editions_, as supported by the latest version of the Rust compiler.
+
+- Once the specification reaches an initial usable version,
+  the relevant teams are expected to incorporate it in their process for language evolution.
+  For example, the language team could require a new language feature to be included
+  in the specification as a requirement for stabilization.
+
+- The specification will be written in English and will be freely available under a permissive license
+  that allows for translations and other derived works, just like all our existing documentation and code.
```

Comment:
I hope this is not intended to exclude parts of the spec being written in formal languages. :) (e.g., like a-mir-formality or MiniRust)

Reply:
No, that's not my intention at all. See also this point later in the RFC:

>  - How certain topics will be specified.
> 
>    Certain parts of the specification might use a formal language for specifying behavior or syntax. For example, the grammar might be specified as EBNF, and parts of the borrow checker or memory model might be specified by a more formal definition that the document refers to.